plugins/modules/identity/keycloak/keycloak_group.py

```diff
@@ -254,112 +255,120 @@ def main():
     name = module.params.get('name')
     attributes = module.params.get('attributes')
 
-    before_group = None         # current state of the group, for merging.
-
-    # does the group already exist?
-    if gid is None:
-        before_group = kc.get_group_by_name(name, realm=realm)
-    else:
-        before_group = kc.get_group_by_groupid(gid, realm=realm)
-
-    before_group = {} if before_group is None else before_group
-
     # attributes in Keycloak have their values returned as lists
     # via the API. attributes is a dict, so we'll transparently convert
     # the values to lists.
     if attributes is not None:
         for key, val in module.params['attributes'].items():
             module.params['attributes'][key] = [val] if not isinstance(val, list) else val
 
+    # Filter and map the parameters names that apply to the group
     group_params = [x for x in module.params
                     if x not in list(keycloak_argument_spec().keys()) + ['state', 'realm'] and
                     module.params.get(x) is not None]
 
-    # build a changeset
-    changeset = {}
+    # See if it already exists in Keycloak
+    if gid is None:
+        before_group = kc.get_group_by_name(name, realm=realm)
+    else:
+        before_group = kc.get_group_by_groupid(gid, realm=realm)
+
+    if before_group is None:
+        before_group = dict()
+
+    # Build a proposed changeset from parameters given to this module
+    changeset = dict()
+
     for param in group_params:
         new_param_value = module.params.get(param)
         old_value = before_group[param] if param in before_group else None
         if new_param_value != old_value:
             changeset[camel(param)] = new_param_value
 
-    # prepare the new group
-    updated_group = before_group.copy()
-    updated_group.update(changeset)
+    # Prepare the desired values using the existing values (non-existence results in a dict that is save to use as a basis)
+    desired_group = before_group.copy()
+    desired_group.update(changeset)
 
-    # if before_group is none, the group doesn't exist.
-    if before_group == {}:
+    # Cater for when it doesn't exist (an empty dict)
+    if before_group == dict():
         if state == 'absent':
-            # nothing to do.
+            # Do nothing and exit
             if module._diff:
                 result['diff'] = dict(before='', after='')
+            result['changed'] = False
+            result['end_state'] = dict()
             result['msg'] = 'Group does not exist; doing nothing.'
-            result['group'] = dict()
```

Comment:
In such cases it's probably best to return info under both names. Eventually we could deprecate the old name, but for now this would break backwards compatibility (and thus cannot be merged).

Reply:
I'll bring it back, (and also update the `RETURN` varibles to make sure they're correct for all the modules I've modified.

I'll be updating this PR accordingly.